File: p1.c
#include <stdio.h>
#define N 1000

int divides(int a, int b) {
    return b != 0 ? a%b == 0 : -1;
}

int main(void)
{
    int i, count = 0;
    int d1 = 3, d2 = 5;
    for(i = 0; i < N; i++)
    {
        if(divides(i, d1) || divides(i,d2))
            count += i;
    }

    printf("Sum of multiples of %d or %d less than %d: %d\n", d1, d2, N, count);
}

File: go/p1.go
package main

import "fmt"

func divides(a, b int) bool {
    return a%b == 0

}

func main() {
    d1, d2 := 3, 5
    sum := 0

    for i := 1; i < 1000; i++ {
        if divides(i, d1) || divides(i, d2) {
            sum += i
        }
    }
    fmt.Printf("Sum of numbers < 1000 that are multiples of 3 and 5: %d\n", sum)
}
